### ISSUE TITLE
About evaluation constraint and energy threshold

### ISSUE THREAD

**Issue by misraya** [deeplearning-wisc/vos#26]
Hi,

I have two questions about the paper and the code. 

1- You say that _"We manually examine the OOD images to ensure they do not contain ID category."_ However, to my understanding, there is no equation in the methodology that requires this constraint. So what is the reason that we can't evaluate on datasets that have common classes with the training dataset?

2- In [apply_net code](https://github.com/deeplearning-wisc/vos/blob/375c05d9ae1ee47fc597a203d4168403bb7ef198/detection/apply_net.py#L97), there is an energy threshold of 8.868 you pass when visualizing and it's used in [this line](https://github.com/deeplearning-wisc/vos/blob/375c05d9ae1ee47fc597a203d4168403bb7ef198/detection/inference/inference_core.py#L131) to label OOD. 

However, I don't see this value being used if the visualization is 0, and I don't see it in the paper either. Can you explain how do you calculate and use this value? If it's in fact used to label OOD, where in the non-visualization code does this happen? Do we need to modify it if we are evaluating on different datasets? If so, how to tune it? 

Thanks a lot in advance!

**Comment by d12306** [comment]
Thanks for your question.

1. the reason is the object detection model will output a series of objects in one image, if your image contains both ID and OOD objects, then how are you gonna get the labels of these objects in terms of whether they belong to ID or OOD? Having separate ID and OOD dataset is the common practice in OOD detection literature, which is different from open-set recognition.
2. the threshold 8.868 is the energy score when the true positive rate of ID samples is at 95%. The model is trained on bdd dataset. You can choose whether you want to visualize the objects or not. I think you need to look at the details of the visualization code.
Thanks,
